lib/sso_users_api/manager.rb

```diff
@@ -12,12 +19,16 @@ def call
     private
 
     def new_sso_user
-      return nil if sso_id
-      return nil if user_exists?
+      return if sso_id
+      return if user_exists?
       response = SsoUsersApi::User.new(base_attrs).create
       # since the sso_id was nil, let's try to update it
-      if user.respond_to?(:sso_id)
+      if user.respond_to?(:sso_id) && response&.ID
         user.update(sso_id: response.ID)
+        NfgRestClient::Logger.info("---SsoUsersApi::Manager---User SSO-ID:#{user.sso_id} has been updated after create.")
+      else
+        # if .create method didn't return an ID with a value then we retry to find the record
```

Comment:
Do you think this should have a spec, i.e. to verify that if no ID is returned, it attempts to look up the user